### PR TITLE
Disable huge pages for every mmap call

### PR DESCRIPTION
**Pull request by adnan-alhomssi**
Disable huge page to avoid THP collapsing ranges of smaller 4 KiB pages into huge pages that can waste lots of memory e.g., when we create lots of tasks.
cf.[ raicode PR](https://github.com/RelationalAI/raicode/pull/14546)

### CODE REVIEW

**Review by li1 — APPROVED** [review]
I don't know what the implications of this are, so perhaps better to wait on Kiran / Nathan / Diogo, but changes themselves LGTM.

**Comment by adnan-alhomssi** [comment]
FYI @NHDaly I have already merged it on raicode so it is probably better to merge it here too for now at least.

**Review by NHDaly — APPROVED** [review]
LGTM. This seems like a good change. Agreed that now that we've merged https://github.com/RelationalAI/raicode/pull/14546, we should merge this, even if there was disagreement, but I think there isn't. :)

--------

The one change that we want to discuss from the Julia side is that we aren't very comfortable with carrying patches in raicode that haven't been proposed in open source julia. We want to make it a prerequisite to first at least open a PR to julia upstream, before merging a julia patch to raicode.

I'll bring that up more in other channels though. For now, this LGTM

**Pull request merged**
